### ISSUE TITLE
[Help]  Unable to auth after upgrading

### ISSUE THREAD

**Issue by aaronfeng** [weaveworks/eksctl#5569]
I recently upgraded from `0.71` to `0.108.0-rc.0`.  I am receiving the following message when trying to list the nodegroups:

```
eksctl get nodegroup --cluster=xxx
Error: unable to describe cluster control plane: operation error EKS: DescribeCluster, https response error StatusCode: 403, RequestID: d96831c8-a16c-424f-8b5c-6c7a10abc04c, api error AccessDeniedException: User: arn:aws:iam::xxxxxxxxxxx:user/xxx@xxx.com is not authorized to perform: eks:DescribeCluster on resource: arn:aws:eks:us-east-1:xxxxxxxxxxxx:cluster/test with an explicit deny
```

However, in the same exact terminal, if I downgrade to `0.71` again, that command works just fine.  I tried a few random version greater than `0.71`, I'm getting the same result.  Any idea what's going on here?  I don't believe this is an IAM issue, since I'm setup as admin of the AWS account.

**Comment by aaronfeng** [comment]
I am able to run

```
aws eks describe-cluster --name xxx
```

**Comment by aaronfeng** [comment]
ok I think I figured out the issue.  I used env vars like `AWS_PROFILE`, `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`.  However it seems (my theory) newer version of `eksctl`  only reads from `~/.aws/credentials` except for `AWS_PROFILE`  from the environment.  Once I added my profile to `~/.aws/credentials` it started to work.  However, this was not the behavior for version `0.71`.
